engine/storage/src/main/java/org/apache/cloudstack/storage/allocator/ZoneWideStoragePoolAllocator.java

```diff
@@ -100,34 +100,41 @@
             if (filter(avoid, storagePool, dskCh, plan)) {
                 suitablePools.add(storagePool);
             } else {
-                avoid.addPool(storagePool.getId());
+                if (isAddStoragePoolToAvoidSet(storage)) {
+                    avoid.addPool(storagePool.getId());
+                }
             }
         }
         return suitablePools;
     }
 
+    // Don't add zone-wide, managed storage to the avoid list because it may be usable for another cluster.
+    private boolean isAddStoragePoolToAvoidSet(StoragePoolVO storagePoolVO) {
```

Comment:
What about `can...` instead of `is...`?
Also, what about a different sentence as the documentation/comment?
`Zone-wide or non-managed storage systems should be added to the avoid list because they might be usable by other clusters`

Reply:
Done